### PR TITLE
Watch later & history

### PR DESCRIPTION
**Pull request by suryxks**
features implemented:
-add videos to history
-add to watch later
-remove from watch later

### CODE REVIEW

File: src/contexts/DataContext.jsx [hunk 1]
```diff
@@ -3,25 +3,103 @@ import axios from 'axios';
 const DataContext = createContext(null);
 
 export const DataProvider = ({ children }) => {
-    const [videos,setVideos]=useState([]);
-    const [err,seterr]=useState('')
-    const getVideos=async ()=>{
+    const [videos, setVideos] = useState([]);
+    const [history, setHistory] = useState([]);
+    const [watchLater, setWatchLater] = useState([]);
+    const [err, seterr] = useState('');
```
Comment (preethasri) [inline src/contexts/DataContext.jsx:9]: just a suggestion you can use reducer instead of use state

**Review by preethasri — APPROVED** [review]
UI looks neat and clean .good to merge

File: src/pages/History/History.jsx [hunk 2]
```diff
@@ -1,11 +1,29 @@
-import { Header, Sidebar } from '../../components/index';
+import { useEffect } from 'react';
+import { Header, Sidebar,VideoCard } from '../../components/index';
+import { useData } from '../../contexts/DataContext';
 import './History.css'
 export const History = () => {
+    const {history,getHistory}=useData();
+    useEffect(()=>{
+        getHistory();
+    },[])
```
Comment (poornima-alugubelly) [inline src/pages/History/History.jsx:9]: not really sure why there is getHistory() inside a useEffect u can directly map thru `history` which would have the history of all the videos user watched other than this all good

**Review by poornima-alugubelly — APPROVED** [review]

File: src/pages/History/History.jsx (hunk 2 above)
Comment (suryxks) [reply src/pages/History/History.jsx:9]: `history` is an empty array and i am getting the previous watch history from server on the history page only as i have not done it in context

**Pull request merged**